### PR TITLE
[Canceled] RedactorWrapper component

### PR DESCRIPTION
**Pull request by sscholle**
## Proposed changes

- Add RedactorWrapper Component
- Add tests
- Add stories

## Types of changes

What type of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] New feature
- [ ] Bugfix
- [ ] Quality of Life

## Checklist

_Put an `x` in the boxes that apply._

- [x] Linter passes (`yarn lint`)
- [x] Unit Test pass (`yarn jest`)
- [x] I have added at least two useful tests for new component (if applicable)

## Relevant Issues
- https://github.com/bonlineza/objectives-and-key-results/issues/18
- https://github.com/bonlineza/objectives-and-key-results/issues/17
- https://github.com/bonlineza/objectives-and-key-results/issues/2

### CODE REVIEW

File: src/components/RedactorWrapper/index.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
+import React from 'react';
+import { canUseRedactor } from '../../helpers/featureDetection';
+import RedactorInit from '../../vendor/redactor/redactor';
+import '../../vendor/redactor/_scss/redactor.scss';
+
+export function getEditorContent(editorId) {
+  if (canUseRedactor()) {
+    // use Redactor
+    return window.$R(`#${editorId}`, 'source.getCode');
```
Comment (robguy21) [inline src/components/RedactorWrapper/index.js]: this should just be a selector so we don't have a requirement of having an id

File: src/components/RedactorWrapper/index.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  }
+
+  componentDidUpdate(prevProps) {
+    if (
+      prevProps.placeholder !== this.props.placeholder &&
+      !this.state.hasContent &&
+      !this.state.legacyBehaviour
+    ) {
+      $R(`#${this.props.editorId}`, 'destroy');
+      $R(`#${this.props.editorId}`, this.settings);
```
Comment (robguy21) [inline src/components/RedactorWrapper/index.js]: yeah this becomes way more simple if we use `this.props.editorSelector`

File: src/components/RedactorWrapper/index.js (hunk 1 above)
Comment (sscholle) [reply src/components/RedactorWrapper/index.js]: what do you mean a 'selector'... not sure how you image it

File: src/components/RedactorWrapper/index.js (hunk 2 above)
Comment (sscholle) [reply src/components/RedactorWrapper/index.js]: plz elaborate

**Comment by sscholle** [comment]
oh dear, will remove it asap - will have to make a lib for private components

**Comment by robguy21** [comment]
> oh dear, will remove it asap - will have to make a lib for private components

I don't think it'll be so bad to let every project handle it this itself.

Redactor is licensed and not every project needs to buy into that license by default.

**Comment by sscholle** [comment]
redactor wrapper must be private due to licence agreement

**Pull request closed without merging**